### PR TITLE
add enhancement with child census vectors and future functionality for tax data

### PR DESCRIPTION
**Pull request by mountainMath**
fixes https://github.com/mountainMath/cancensus/issues/134 and adds support for future tax data release

### CODE REVIEW

File: R/census_vectors.R [hunk 1]
```diff
@@ -0,0 +1,221 @@
+#' Query the CensusMapper API for available vectors for a given dataset.
```
Comment (dshkol) [inline R/census_vectors.R]: Which part adds support for the tax releases?

Comment (mountainMath) [reply R/census_vectors.R]: New aggregation type and units in list_census_vectors to help with new metadata and variable types.

File: R/census_vectors.R [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' list_census_vectors("CA16") %>%
+#'   filter(vector == "v_CA16_4092") %>%
+#'   parent_census_vectors()
+parent_census_vectors <- function(vector_list){
```
Comment (dshkol) [inline R/census_vectors.R]: This should try to work the same way as child census vectors. Right now:

```
list_census_vectors("CA16") %>%
  filter(vector == "v_CA16_4092") %>%
  parent_census_vectors()
```
Works as usual, but
```
  parent_census_vectors("v_CA16_4092")
```
Throws a very odd error
```
Error in handle_cm_status_code(response, NULL) : 
  Download of Census Data failed. <!DOCTYPE html>
<html>
<head>
  <title>The page you were looking for doesn't exist (404)</title>
  <meta name="viewport" content="width=device-width,initial-scale=1">
  <style>
  body {
    background-color: #EFEFEF;
    color: #2E2F30;
    text-align: center;
    font-family: arial, sans-serif;
    margin: 0;
  }

  div.dialog {
    width: 95%;
    max-width: 33em;
    margin: 4em auto 0;
  }

  div.dialog > div {
    border: 1px solid #CCC;
    border-right-color: #999;
    border-left-color: #999;
    border-bottom-color: #BBB;
    border-top: #B00100 solid 4px;
    border-top-left-radius: 9px;
    border-top-right-radius: 9px;
    background-color: white;
    padding: 7px 12% 0;
    box-shadow: 0 3px 8px rgba(50, 50, 50, 0.17);
  }

  h1 {
    font-size: 100%;
    color: #730E15;
    line-height: 1.5em;
  }

  div.dialog > p {
    margin: 0 0 1em;
    padding: 1em;
    background-color: #F7F7F7;
    border: 1px solid #CCC;
    border-right-
```
If we don't want the same functionality, should at least add better error handling here.

File: R/census_vectors.R [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' list_census_vectors("CA16") %>%
+#'   filter(vector == "v_CA16_4092") %>%
+#'   child_census_vectors(TRUE)
+child_census_vectors <- function(vector_list, leaves_only=FALSE,max_level=NA){
```
Comment (dshkol) [inline R/census_vectors.R]: Tested both old and new method here, and works nicely. The examples need to be updated though with the new functionality, also that particular vector doesn't have children oops

File: R/census_vectors.R (hunk 2 above)
Comment (mountainMath) [reply R/census_vectors.R]: good catch, did not update the parent_census_vectors call. will do that.

**Comment by mountainMath** [comment]
Ok, fixed the parent_census_vectors call. And moved the census_regions stuff into a separate file to make things cleaner. Also changed the caching behaviour of list_census_regions and list_census_vectors to cache for duration of running R session by default. Should make things run smoother.

**Comment by dshkol** [comment]
Added some tweaks to the documentation and examples that were bothering me or were in need of an update for new syntax.

**Comment by dshkol** [comment]
If there's no CRAN check issues or notes (I had one, but it seemed specific to my local directory), happy to push this one into master and move to submit the update to CRAN

**Comment by mountainMath** [comment]
CRAN check looks good on my end. Only other thing I can think of is adding tests. Maybe leave that for the next release? 😁

**Comment by dshkol** [comment]
Should be added, but I think can wait for the next one.

**Comment by dshkol** [comment]
Let's merge and submit then?

**Pull request merged**